docs/designers-developers/developers/block-api/block-edit-save.md

```diff
@@ -80,6 +80,23 @@ edit( { attributes, setAttributes, className, isSelected } ) {
 }
 ```
 
+When calling `setAttributes`, it's important that attributes are not mutated in the process of updating them. This is considered bad practice, the core of Gutenberg has been written with the same philosophies as Redux and state is treated as immutable data. It can also cause bugs. Objects and arrays are passed as references in JavaScript, so mutating them can affect other bindings to those references, for example an attribute's default value.
```

Comment:
This makes a lot of sense to document but I think the lingo might be slightly over the head of many WP developers.

It's obviously good to introduce and use higher level terms so folks can learn and expand their skills, but maybe this can be expanded with some extra context, definitions, etc.?

(This could also link into @mkaz's work on a JS tutorial, so we could point someone to some additional background.)

Reply:
Thanks for the feedback @chrisvanpatten—I've simplified the text a bit, tried using a simpler vocabulary.